### PR TITLE
Expanding on ways of working guide

### PR DESCRIPTION
**Pull request by ewastempel**
This change is part of the [#2641](https://github.com/ministryofjustice/modernisation-platform/issues/2641) and:
1. Introduces working with Git/GitHub section
2. Updates task management section

This has been tested locally.

### CODE REVIEW

File: source/team/ways-of-working.html.md.erb [hunk 1]
```diff
@@ -24,7 +28,7 @@ As a team we have agreed to do the following for each stage:
     * Tasks are roughly prioritised during a sprint planning session, but we are not obliged to take the one from the top.
     We have the freedom of picking a task that we want to work on and we do not pre-assign team members to a task prior the start of a sprint.
     * Tasks that are selected for a sprint should have all the necessary information in the task description, including a Definition of Done (DoD).
-    If it is not clear what is asked in the task, it does not meet the Definition of Ready (DoR) and should be taken out of the sprint.
+    If it is not clear what is asked in the task, then it does not meet the DoR and therefore should be taken out of the sprint.
```
Comment (jakemulley) [inline source/team/ways-of-working.html.md.erb:31]: Could you add back in the explanation of what DoR means? I've not recently seen DoR used before, so was slightly confused.

File: source/team/ways-of-working.html.md.erb [hunk 2]
```diff
@@ -38,3 +42,24 @@ As a team we have agreed to do the following for each stage:
...
+## Working with Git and GitHub
+
+There is a number of things we can do to make collaborating on code seamless.
+Although we do not want to impose too many rules on our team members, the following list is what we agreed as a team to do as a minimum:
+
+* Keeping a good git hygiene `git fetch`/`git pull` in `main` before branching out when starting on a new task. Merging `main` regularly into your branch on local and resolving the merge conflicts before committing and pushing to remote.
+* Adding meaningful messages when committing, such as what have you achieved in this step.
+* Running tests, verifying if the tests/SCA has passed and that the `terraform plan` and the pipeline runs clean before creating a PR.
+* Providing enough information in the PR description so that another team member has enough context to be able to understand the change and to review it.
+Adding explanation on what is implemented with the PR and why. Adding information on tests if they were run outside of the GitHub actions pipeline and other useful links or references that help to understand the changes (e.g. a link to the issue or a doc).
```
Comment (jakemulley) [inline source/team/ways-of-working.html.md.erb:55]: Is this line missing a `*`?

File: source/team/ways-of-working.html.md.erb (hunk 1 above)
Comment (ewastempel) [reply source/team/ways-of-working.html.md.erb:31]: it is explained earlier on in the doc, hence I removed it

File: source/team/ways-of-working.html.md.erb (hunk 2 above)
Comment (ewastempel) [reply source/team/ways-of-working.html.md.erb:55]: it's a continuation of the previous line

**Review by jakemulley — APPROVED** [review]

**Pull request merged**